src/utilities/utilities.jl

```diff
@@ -299,3 +299,26 @@ function attribute_names(PlotType)
     # TODO, have all plot types store their attribute names
     return keys(default_theme(nothing, PlotType))
 end
+
+
+"""
+    regularly_spaced_array_to_range(arr)
+If possible, converts `arr` to a range.
+If not, returns array unchanged.
+"""
+function regularly_spaced_array_to_range(arr)
+    diffs = unique!(sort!(diff(arr)))
```

Comment:
this seems  a bit weird, but maybe it's floating point error avoiding magic ;) I think something like this would be simpler:

```julia
distance = last(arr) - first(arr)
step = distance / (length(arr) - 1)
# iterate through pairs and compare each diff to step, if they all match return range from first to last element
```

I don't know why extrema are being taken and vectors full of diffs being allocated, shouldn't any valid array be already sorted? so we don't need to do the weird min, max switch either

Reply:
This function is just getting moved in this PR and wasn't written by me. If you think it should be written differently, we should make a new PR ;)